### PR TITLE
Add click.Choice options to help text

### LINKED ISSUE #35: Document `click.Choice` options
When you have `type=click.Choices(['a', 'b', 'c'])` in a Click option, it would be very helpful to include those options in the generated help output.

For example, the auto-generated `--help` output has

```
  --log-level [NOTSET|DEBUG|INFO|WARNING|ERROR|FATAL]
```

but the generated HTML just says

> `--log-level <log_level>`

### PR DESCRIPTION
**Pull request by AWegnerGitHub**
If an option has `type=click.Choice`, this will extract the options
and place it at the end of the help text.

It will look like this:

image

Also adds a test case for this.

This should resolve #35 and #41

### CODE REVIEW

**Comment by stephenfin** [comment]
Thanks! Once idea inline. If you can tackle that (or argue against it), I'm happy to merge this

**Comment by AWegnerGitHub** [comment]
I'm sorry, I don't understand what you are asking. Can you rephrase it for me?

**Comment by stephenfin** [comment]
Instead of simply appending the options to the help text, I'm suggesting you add an empty line and then field list (`:foo: bar`) listing the options. I think it would be cleaner looking

**Comment by AWegnerGitHub** [comment]
I have a few options for you:

**Option 1**

Use a custom admonition and put all options into a single string. It'd look like this:

image

The code being generated looks like this: 

```
.. admonition:: Available Options
    aws,azure,google
```
---

**Option 2**

Use a custom admonition but bullet the options. It'd look like this:

image

The generated code looks like this:

```
.. admonition:: Available Options
    - aws
    - azure
    - google
```
---

**Option 3**

Use a field list, as you suggested, and put all options into a single string:

image

This generates code that looks like this:

```
:options: aws|azure|google
```

---

**Option 4**

Use a field list, but bullet each of the options:

image

This generates the following code:

```
:options: 
- aws
- azure
- google
```

---

Personally, I like option 2 (the custom admonition with a bulleted list). This keeps the available options nicely grouped in a box, bulleted for readability and labeled that these are the available options.

However, if you don't like the admonition, I like Option 4 second best. This looks similar to a list of `:param:` fields in the docstrings.

Let me know which you prefer and I'll update my PR.

**Comment by stephenfin** [comment]
> Personally, I like option 2 (the custom admonition with a bulleted list). This keeps the available options nicely grouped in a box, bulleted for readability and labeled that these are the available options.
> 
> However, if you don't like the admonition, I like Option 4 second best. This looks similar to a list of `:param:` fields in the docstrings.

I like option 3, personally, since it's the most compact and I don't think choices are so important that they need huge emphasis. After that, option 1 would be my next favourite followed by option 4. Is option 3 okay for you? If not, maybe option 1

**Comment by AWegnerGitHub** [comment]
Option 3 is good. :+1:  I'll get my PR updated in a few hours with that change.

**Comment by AWegnerGitHub** [comment]
I believe this is ready to go

**Comment by stephenfin** [comment]
:clap:
